app/src/main/java/edu/uc/hickmadc/customnotifications/ui/main/dialogfragment.kt

```diff
@@ -5,30 +5,58 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Button
+import android.widget.ImageButton
+import android.widget.TextView
 import androidx.fragment.app.DialogFragment
 import androidx.fragment.app.FragmentContainer
 import edu.uc.hickmadc.customnotifications.R
+import edu.uc.hickmadc.customnotifications.dto.Notification
 import java.io.FileDescriptor
 import java.io.PrintWriter
 import java.util.zip.Inflater
 
 
+class dialogfragment : DialogFragment() {
 
-class dialogfragment: DialogFragment() {
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
-    ): View?{
+    private lateinit var rootView: View
+    private lateinit var notificationTitleText: TextView
+    private lateinit var notificationSubtext: TextView
+    private lateinit var notificationText: TextView
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View? {
         //gets dialog fragment
-        var rootView: View= inflater.inflate(R.layout.notification_dialog,container,false)
+        rootView = inflater.inflate(R.layout.notification_dialog, container, false)
+
+        notificationTitleText = rootView.findViewById(R.id.txtTitle)
+        notificationSubtext = rootView.findViewById(R.id.txtSubtext)
+        notificationText = rootView.findViewById(R.id.notificationText)
+
+        addEventHandlersToDialogBox()
+
+        return rootView
+    }
+
+    private fun addEventHandlersToDialogBox() {
+
         //gets ids for buttons
-        var btnClose= rootView.findViewById<Button>(R.id.closeButton)
-        var btnadd = rootView.findViewById<Button>(R.id.btnAdd)
+        var btnClose = rootView.findViewById<ImageButton>(R.id.closeButton)
+        var btnConfirm = rootView.findViewById<Button>(R.id.btnconfirm)
         btnClose.setOnClickListener {
             dismiss()
         }
 
+        btnConfirm.setOnClickListener {
 
-        return rootView
-
+            var notificationToAdd: Notification = Notification(
+                notificationTitleText.text.toString(),
+                notificationSubtext.text.toString(),
+                notificationText.text.toString(),
+                0
+            )
+        }
 
     }
```

Comment:
I extracted your click listener logic out of the onCreateView method into its separate method.  In addition, I added a bit of logic where I retrieve the information the user entered in the dialog box and store it in a notification object. Your next step from here is presumably storing this notification object in your room database.

Reply:
Good.